route.c

```diff
@@ -1059,15 +1060,25 @@ consider_route(struct babel_route *route)
     if(installed == NULL)
         goto install;
 
-    if(route_metric(route) >= INFINITY)
+    if(route_metric(route) >= BABEL_INFINITY)
         return;
 
-    if(route_metric(installed) >= INFINITY)
+    if(route_metric(installed) >= BABEL_INFINITY)
         goto install;
 
 
-    installed_sum_metric = installed->price + route_smoothed_metric(installed) * quality_multiplier;
-    route_sum_metric = route->price + route_smoothed_metric(route) * quality_multiplier;
+    /**
+     * Quick overview of Althea price/quality decision-making:
+     *
+     * Formula:
+     * metric(price, interference, coefficient) = log2(price) + log2(interference) * coefficient
+     *
+     * The gist of the formula is that at coefficient equal to 1 we'll only
+     * install an x times pricier route if it offers *more than x* times lower
+     * interference. i.e. 
+     */
+    installed_sum_metric = log2((double)installed->price) + log2((double)route_smoothed_metric(installed)) * ((double)metric_factor / 1000.0);
+    route_sum_metric = log2((double)installed->price) + log2((double)route_smoothed_metric(route)) * ((double)metric_factor / 1000.0);
```

Comment:
oh we invert the factor, I suppose this saves us the trouble of parsing floats, kinda an odd way to do it, hmm does it really work well? I'll have to think about it some.

Reply:
I made the setting to be a fixed-point decimal value where a 1.0 is represented by a 1000